tests/Codec/LazyBSONCodecLibraryTest.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+namespace MongoDB\Tests\Codec;
+
+use Generator;
+use MongoDB\BSON\Document;
+use MongoDB\BSON\PackedArray;
+use MongoDB\Codec\LazyBSONCodecLibrary;
+use MongoDB\Model\LazyBSONArray;
+use MongoDB\Model\LazyBSONDocument;
+use MongoDB\Tests\TestCase;
+
+class LazyBSONCodecLibraryTest extends TestCase
+{
+    public static function provideDecodedData(): Generator
+    {
+        $array = [
+            'bar',
+            ['foo' => 'bar'],
+            [0, 1, 2],
+        ];
+        $document = (object) [
+            'string' => 'bar',
+            'document' => ['foo' => 'bar'],
```

Comment:
Should you have a nested document here?
```suggestion
            'document' => (object) ['foo' => 'bar'],
```

Reply:
Yes, updated.